_posts/2020-07-31-fnn-vae-for-noisy-timeseries/fnn_vae_for_noisy_timeseries.Rmd

```diff
@@ -0,0 +1,506 @@
+knitr::include_graphics("images/noise_2.5_nofnn.png")
+```
+
+Both LSTM and VAE get "distracted" a bit too much by the noise, the latter to an even higher degree. This leads to cases
+where predictions strongly "overshoot" the underlying non-noisy rhythm. This is not surprising, of course: They were *trained*
+on the noisy version; predict fluctuations is what they learned.
+
+Do we see the same with the FNN models?
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Roessler series with added Gaussian noise of standard deviation 2.5. Grey: actual (noisy) test data. Green: underlying Roessler system. Orange: Predictions from FNN-LSTM. Dark blue: Predictions from FNN-VAE."}
+knitr::include_graphics("images/noise_2.5.png")
+```
+
+Interestingly, we see a much better fit to the underlying Roessler system now! Especially the VAE model, FNN-VAE, surprises
+with a whole new smoothness of predictions; but FNN-LSTM turns up much smoother forecasts as well.
+
+"Smooth, fitting the system..." -- by now you may be wondering, when are we going to come up with more quantitative
+assertions? If quantitative implies "mean squared error" (MSE), and if MSE is taken to be some divergence between forecasts
+and the true target from the test set, the answer is that this MSE doesn't differ much between any of the four architectures.
+Put differently, it is mostly a function of noise level.
+
+However, we could argue that what we're really interested in is how well a model forecasts the underlying process. And there,
+we see differences.
+
+In the following plot, we contrast MSEs obtained for the four model types (grey: VAE; orange: LSTM; dark blue: FNN-VAE; green:
+FNN-LSTM). The rows reflect noise levels (1, 1.5, 2, 2.5); the columns represent MSE in relation to the noisy("real") target
+(left) on the one hand, and in relation to the underlying system on the other (right). For better visibility of the effect,
+*MSEs have been normalized as fractions of the maximum MSE in a category*.
+
+So, if we want to predict *signal plus noise* (left), it doesn't really matter whether we use FNN or not. But if we want to
+predict the signal only (right), with increasing noise in the data FNN loss becomes increasingly effective. This effect is far
+stronger for VAE vs. FNN-VAE than for LSTM vs. FNN-LSTM: The distance between the grey line (VAE) and the dark blue one
+(FNN-VAE) becomes larger and larger as we add more noise.
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Normalized MSEs obtained for the four model types (grey: VAE; orange: LSTM; dark blue: FNN-VAE; green: FNN-LSTM). Rows are noise levels (1, 1.5, 2, 2.5); columns are MSE as related to the real target (left) and the underlying system (right)."}
```

Comment:
It took me some time to clearly understand the plot, I had to come back and forth to the text multiple times to remember what's in each column/row. Also there's no description of what's the `x` axis.. I assumed that it's the prediction step.

Reply:
Oh +1 on this—I usually proof in Rmd, so I hadn't looked at the preview, but I'd definitely add a legend (check out [patchwork](https://github.com/thomasp85/patchwork) or [cowplot](https://github.com/wilkelab/cowplot), maybe, if you want finer-tuned control). Let me know if you need any help!

Reply:
i'm using cowplot ... i didn't use any legends anywhere, to save space, as anyway i'm always having the text below the figures....

Reply:
changed the text to say what's on the x axis